src/System.Private.CoreLib/shared/System/Activator.RuntimeType.cs

```diff
@@ -132,26 +132,122 @@ internal static object CreateInstance(Type type, bool nonPublic, bool wrapExcept
+                }
+
+                var instance = default(T);
+
+                if (ActivatorCache<T>.ConstructStruct != null)
+                {
+                    try
+                    {
+                        ActivatorCache<T>.ConstructStruct(ref instance);
+                    }
+                    catch (Exception exception) when (wrapExceptions)
+                    {
+                        throw new TargetInvocationException(exception);
+                    }
+                }
+
+                return instance;
+            }
+            else
+            {
+                if (ActivatorCache<T>.HasNonPublicConstructor &&
+                    ActivatorCache<T>.ConstructClass != null &&
+                    publicOnly)
+                {
+                    throw new MissingMethodException(SR.Format(SR.Arg_NoDefCTor, ActivatorCache<T>.Type));
+                }
+
+
+                var instance = (T)RuntimeTypeHandle.Allocate(ActivatorCache<T>.Type);
 
-            // Skip the CreateInstanceCheckThis call to avoid perf cost and to maintain compatibility with V2 (throwing the same exceptions).
-            return (T)rt.CreateInstanceDefaultCtor(publicOnly: true, skipCheckThis: true, fillCache: true, wrapExceptions: true);
+                if (ActivatorCache<T>.ConstructClass != null)
+                {
+                    try
+                    {
+                        ActivatorCache<T>.ConstructClass(instance);
+                    }
+                    catch (Exception exception) when (wrapExceptions)
+                    {
+                        throw new TargetInvocationException(exception);
+                    }
+                }
+
+                return instance;
+            }
+        }
+
+        private delegate void ConstructorOfClass<T>(T instance);
+        private delegate void ConstructorOfStruct<T>(ref T instance);
+
+        private static T CreateDelegate<T>(MethodBase methodInfo)
+            where T : Delegate
+        {
+            return (T)typeof(T)
+               .GetConstructor(new[] { typeof(object), typeof(IntPtr) })
+               .Invoke(new object[] { null, methodInfo.MethodHandle.GetFunctionPointer() });
+        }
+
+        private static class ActivatorCache<T>
+        {
+
+            internal static readonly RuntimeType Type;
+            internal static readonly bool IsValueType;
+            internal static readonly bool HasElementType;
+            internal static readonly bool HasNonPublicConstructor;
+
+            internal static readonly ConstructorOfClass<T> ConstructClass;
+            internal static readonly ConstructorOfStruct<T> ConstructStruct;
+
+            static ActivatorCache()
+            {
+                Type = (RuntimeType)typeof(T);
+                IsValueType = Type.IsValueType;
+                HasElementType = Type.HasElementType;
+
+                if (!HasElementType)
```

Comment:
A logical bug was made here. It prevented the constructor invocation and therefore led to NRE.